monitor.py

```diff
@@ -880,43 +881,54 @@ def parse_entry(self, line: bytes) -> MutableMapping[str, Any]:  # noqa: C901, C
             elif event_type in ('BackPack', 'Backpack'):  # WORKAROUND 4.0.0.200: BackPack becomes Backpack
                 # TODO: v31 doc says this is`backpack.json` ... but Howard Chalkley
                 #       said it's `Backpack.json`
-                with open(join(self.currentdir, 'Backpack.json'), 'rb') as backpack:  # type: ignore
-                    try:
-                        # Preserve property order because why not?
-                        entry = json.load(backpack, object_pairs_hook=OrderedDict)
+                backpack_file = pathlib.Path(str(self.currentdir)) / 'Backpack.json'
+                backpack_data = None
+                if backpack_file.exists():
+                    backpack_data = backpack_file.read_bytes()
+                else:
+                    logger.warning(f'Failed to find backpack.json file as it appears not to exist? {backpack_file=}')
 
+                parsed = None
+
+                if backpack_data is not None and len(backpack_data) > 0:
+                    try:
+                        parsed = json.loads(backpack_data)
                     except json.JSONDecodeError:
-                        logger.exception('Failed decoding Backpack.json', exc_info=True)
+                        logger.exception('Unable to parse Backpack.json')
 
-                    else:
-                        # Store in monitor.state
-                        self.state['BackpackJSON'] = entry
-
-                        # Assume this reflects the current state when written
-                        self.state['BackPack']['Component'] = defaultdict(int)
-                        self.state['BackPack']['Consumable'] = defaultdict(int)
-                        self.state['BackPack']['Item'] = defaultdict(int)
-                        self.state['BackPack']['Data'] = defaultdict(int)
-
-                        clean_components = self.coalesce_cargo(entry['Components'])
-                        self.state['BackPack']['Component'].update(
-                            {self.canonicalise(x['Name']): x['Count'] for x in clean_components}
-                        )
-
-                        clean_consumables = self.coalesce_cargo(entry['Consumables'])
-                        self.state['BackPack']['Consumable'].update(
-                            {self.canonicalise(x['Name']): x['Count'] for x in clean_consumables}
-                        )
-
-                        clean_items = self.coalesce_cargo(entry['Items'])
-                        self.state['BackPack']['Item'].update(
-                            {self.canonicalise(x['Name']): x['Count'] for x in clean_items}
-                        )
-
-                        clean_data = self.coalesce_cargo(entry['Data'])
-                        self.state['BackPack']['Data'].update(
-                            {self.canonicalise(x['Name']): x['Count'] for x in clean_data}
-                        )
+                else:
```

Comment:
I'd prefer splitting the conditional to separately log each case.

One is "Frontier said there was a file, there wasn't".
Another is "Frontier said there was a file, there was, but it was empty".
Another is "Frontier said there was a file, there was, it had content, but it didn't parse as JSON".